src/components/AddActivityPage.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import AcivityForm from './ActivityForm';
+import { addActivity } from '../actions/activities';
+
+const AddActivityPage = (props) => (
+  <div>
+    <h1>Add Activity</h1>
+    <AcivityForm
+      onSubmit={(activity) => {
+        props.dispatch(addActivity(activity));
+        props.history.push('/');
+      }}
+    />
+  </div>
+);
+
+export default connect()(AddActivityPage);
```

Comment:
whhy are you connecting this file to store?

Reply:
ah.. I see. 
You should do:
```
const mapDispatchToProps = {
   onAddActivity: addActivity`
}

export default connect(null, mapDispatchh...`

and then use as an action `onAddActivity` in component (with props destructured) (or name it as you wish, but similar to original action).

Reply:
we need to talk about this, just to clarify

Reply:
not forget :P